### PR TITLE
Fix macOS section handling that already have a comma in ther name

### PR DESCRIPTION
**Pull request by ptitSeb**
## Description

Following the commit [#3917e32](https://github.com/TheDan64/inkwell/commit/3917e3207941c1a90f28df4fe08ffdd26de71575) section have a `,` being added to their name on macOS. On some certain cases, that `,` should not be added because the name is already fully qualified, like `__TEXT,wasmer_function`.
This PR check if section name already contains a comma before adding it.

## Related Issue

address issue #396 

## How This Has Been Tested

Tested with latest Wasmer on macOS , both x86_64 and aarch64 arch.

## Checklist

- [x] I have read the [Contributing Guide](https://github.com/TheDan64/inkwell/blob/master/.github/CONTRIBUTING.md)

### CODE REVIEW

File: src/values/mod.rs [hunk 1]
```diff
@@ -195,7 +202,7 @@ impl<'ctx> Value<'ctx> {
     /// Sets the section of the global value
     fn set_section(self, section: Option<&str>) {
         #[cfg(target_os = "macos")]
-        let section = section.map(|s| format!(",{}", s));
+        let section = section.map(|s| if s.contains(",") { format!("{}", s) } else { format!(",{}", s) });
```
Comment (TheDan64) [inline src/values/mod.rs:205]: Should this be starts_with?

Comment (ptitSeb) [reply src/values/mod.rs:205]: No. the idea is if a section name is just "mystuff", it needs to be set as ",mystuff", but if it's "XXX,mystuff", then don't touch it.
Section are expected to be in the format "<segment name>,<section name>", and it seems a no segment-name is acceptable, that why the previous commit added the ",".

**Comment by syrusakbary** [comment]
@TheDan64 is there any blocker for merging the PR? We would love to use it

**Comment by TheDan64** [comment]
I can take a look at merging this and releasing a v0.1.1 this weekend

**Review by TheDan64 — APPROVED** [review]

**Comment by BigBadE** [comment]
Does this PR even build? This shadows "section" to be an Option<String> instead of an Option<&str>, which can't be passed to to_c_str.

**Pull request merged**